x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/crossclusteraccess/CrossClusterAccessAuthenticationServiceIntegTests.java

```diff
@@ -135,16 +146,186 @@ public void testInvalidHeaders() throws IOException {
+            tryAuthenticateAndAssertExpectedErrorMessage(
+                service,
+                threadContext,
+                withRandomizedAdditionalSecurityHeaders(Map.of()),
+                msg -> assertThat(
+                    msg,
+                    containsString(
+                        "Cross cluster requests through the dedicated remote cluster server port require transport header ["
+                            + CROSS_CLUSTER_ACCESS_CREDENTIALS_HEADER_KEY
+                            + "] but none found. "
+                            + "Please ensure you have configured remote cluster credentials on the cluster originating the request."
+                    )
+                )
+            );
+        }
+
+        try (var ignored = threadContext.stashContext()) {
+            addRandomizedHeaders(threadContext, encodedCrossClusterAccessApiKeyWithId.encoded);
+            tryAuthenticateAndAssertExpectedErrorMessage(
+                service,
+                threadContext,
+                withRandomizedAdditionalSecurityHeaders(
+                    Map.of(CROSS_CLUSTER_ACCESS_CREDENTIALS_HEADER_KEY, ApiKeyService.withApiKeyPrefix("abc"))
+                ),
+                msg -> assertThat(
+                    msg,
+                    containsString(
+                        "cross cluster access header ["
+                            + CROSS_CLUSTER_ACCESS_CREDENTIALS_HEADER_KEY
+                            + "] value must be a valid API key credential"
+                    )
+                )
+            );
+        }
+
+        try (var ignored = threadContext.stashContext()) {
+            addRandomizedHeaders(threadContext, encodedCrossClusterAccessApiKeyWithId.encoded);
+            Map<String, String> headers = withRandomizedAdditionalSecurityHeaders(
+                Map.of(CROSS_CLUSTER_ACCESS_CREDENTIALS_HEADER_KEY, encodedRestApiKeyWithId.encoded)
+            );
+            final PlainActionFuture<Void> future = new PlainActionFuture<>();
+            service.doTryAuthenticateCredentialsHeader(threadContext, headers, future);
+            final ExecutionException actualException = expectThrows(ExecutionException.class, future::get);
+            assertThat(actualException.getCause(), instanceOf(ElasticsearchSecurityException.class));
+            assertThat(
+                actualException.getCause().getMessage(),
+                containsString("authentication expected API key type of [" + ApiKey.Type.CROSS_CLUSTER.value() + "]")
+            );
+        }
+
+        try (var ignored = threadContext.stashContext()) {
+            addRandomizedHeaders(threadContext, encodedCrossClusterAccessApiKeyWithId.encoded);
+            final String wrongApiKeyWithCorrectId = ApiKeyService.withApiKeyPrefix(
+                Base64.getEncoder()
+                    .encodeToString((UUIDs.base64UUID() + ":" + UUIDs.randomBase64UUIDSecureString()).getBytes(StandardCharsets.UTF_8))
```

Comment:
Nit: Isn't this `wrongApiKeyWithIncorrectId` since the ID does not exist?

Reply:
Ah crap copy and paste. I had another test with the correct ID that somehow did not survive a refactoring. Thanks for catching 👍